lms/djangoapps/mobile_api/tests/test_course_dates_views.py

```diff
@@ -58,110 +43,60 @@ def setUp(self):
             field="due",
             policy=self.date_policy_1,
         )
-        self.content_date_without_enrollment = ContentDate.objects.create(
-            course_id=self.course_without_enrollment.id,
-            location=self.sequential_without_enrollment.location,
-            active=True,
-            block_type="sequential",
-            field="due",
-            policy=self.date_policy_1,
-        )
 
-    def _get_view_with_request(self):
-        """
-        Returns an instance of the view with a request.
-        """
-        request = APIRequestFactory().get(f"/api/mobile/v1/course_dates/{self.user.username}/")
-        view = AllCourseDatesAPIView()
-        view.request = request
-        view.kwargs = {"username": self.user.username}
-        return view
+    def tearDown(self):
+        UserDate.objects.all().delete()
+        super().tearDown()
 
-    def test_get_queryset(self):
+    def test_only_future_dates_are_returned(self):
         """
-        Simple test to check if the queryset is returning the correct course dates.
-        Dates from not enrolled courses should not be returned.
+        Ensure GET only returns user dates strictly after now().
         """
-        view = self._get_view_with_request()
-
-        queryset = view.get_queryset()
-        self.assertEqual(queryset.count(), 1)
-        self.assertEqual(queryset.first(), self.content_date_1)
-
-    def test_get_queryset_with_relative_date(self):
+        now = timezone.now()
+        # past date
+        UserDate.objects.create(user=self.user, content_date=self.content_date_1, abs_date=now - timedelta(days=1))
+        # today (should be excluded)
+        UserDate.objects.create(user=self.user, content_date=self.content_date_1, abs_date=now)
+        # future date (should be included)
+        future_date = UserDate.objects.create(user=self.user, content_date=self.content_date_1,
+                                              abs_date=now + timedelta(days=2))
+
+        self.login_and_enroll(self.course_1.id)
+        response = self.api_response()
+
+        results = response.data["results"]
+        self.assertEqual(len(results), 1)
```

Comment:
Should it return 2 items since the `now = timezone.now()` in views.py is executed after the test setup code?